### ISSUE TITLE
[BUG] Extension Initialization invalidates opensearch host/port configurations in the extension.yml

### ISSUE THREAD

**Issue by joshpalis** [opensearch-project/opensearch-sdk-java#769]
### What is the bug?
It is necessary to provide an SDKClient/cluster service for use in integration testing for extensions, similar how the `OpenSearchIntegTestCase` class provides a node client and cluster service [here](https://github.com/opensearch-project/OpenSearch/blob/471060f9dd98a086b81dfaba53a0174e121757df/test/framework/src/main/java/org/opensearch/test/OpenSearchIntegTestCase.java#LL654C57-L654C57). This allows integration tests, such as [AnomalyDetectorRestApiIT](https://github.com/opensearch-project/anomaly-detection/blob/main/src/test/java/org/opensearch/ad/rest/AnomalyDetectorRestApiIT.java) to communicate with the integration test cluster and test the behavior of various rest APIs provided by the extension. 

The `SDKRestClient` is initialized by the host and port configurations of the `extension.yml` [here](https://github.com/opensearch-project/opensearch-sdk-java/blob/066158647667ee7d9f4e1f374426d8603fa198ca/src/main/java/org/opensearch/sdk/SDKClient.java#LL272C123-L272C123), which directs rest requests back to the opensearch node. This usually works quite well, however there are a few use cases in which this  extension initialization process invalidates the extension.yml configuration and makes it impossible to configure a new `SDKRestClient` for use in integration testing.

For example, spinning up an integration test cluster uses an OpenSearch distribution that randomly sets its host and port, irrespective of the host and port listed in the `extension.yml`. This does not cause any issues with the initial handshake with the extension, as OpenSearch uses an `extensions.yml` to identify where the extension is and the `SDKClient` updates it's host and port configurations with the opensearch node host/address after initialization [here](https://github.com/opensearch-project/opensearch-sdk-java/blob/066158647667ee7d9f4e1f374426d8603fa198ca/src/main/java/org/opensearch/sdk/handlers/ExtensionsInitRequestHandler.java#LL67C13-L67C13). However, creating an `SDKClient` for use in integration tests is only possible using the host and port listed in the `extension.yml` and is not updated by the handshake, resulting in the following connection refused errors :

OpenSearch host and port configuration of the `extension.yml` :
```
extensionName: ad-extension
hostAddress: 127.0.0.1
hostPort: 4532
opensearchAddress: 127.0.0.1
opensearchPort: 9200
```
Bound addresses of the OpenSearch node :
```
[integTest-0] publish_address {127.0.0.1:41109}, bound_addresses {[::1]:36075}, {127.0.0.1:41109}
```
Initializing an `SDKClient` with the host and port values of the `extension.yml` results in connection refused error
```
> Task :integTest

REPRODUCE WITH: ./gradlew ':integTest' --tests "org.opensearch.ad.rest.AnomalyDetectorRestApiIT.testDeleteAnomalyDetectorWithRunningAdJob" -Dtests.seed=1E6EA8F2B87F7ADF -Dtests.security.manager=false -Dtests.locale=de-LU -Dtests.timezone=Etc/GMT+8 -Druntime.java=17

org.opensearch.ad.rest.AnomalyDetectorRestApiIT > testDeleteAnomalyDetectorWithRunningAdJob FAILED
    java.net.ConnectException: Connect to http://127.0.0.1:9200 [/127.0.0.1] failed: Connection refused
        at org.opensearch.client.RestClient.extractAndWrapCause(RestClient.java:1035)
        at org.opensearch.client.RestClient.performRequest(RestClient.java:341)
        at org.opensearch.client.RestClient.performRequest(RestClient.java:329)
        at org.opensearch.sdk.SDKClient$SDKRestClient.performRequest(SDKClient.java:549)
        at org.opensearch.ad.TestHelpers.makeRequest(TestHelpers.java:221)
        at org.opensearch.ad.AnomalyDetectorRestTestCase.createRandomAnomalyDetector(AnomalyDetectorRestTestCase.java:97)
        at org.opensearch.ad.AnomalyDetectorRestTestCase.createRandomAnomalyDetector(AnomalyDetectorRestTestCase.java:77)
        at org.opensearch.ad.AnomalyDetectorRestTestCase.createRandomAnomalyDetector(AnomalyDetectorRestTestCase.java:68)
        at org.opensearch.ad.rest.AnomalyDetectorRestApiIT.testDeleteAnomalyDetectorWithRunningAdJob(AnomalyDetectorRestApiIT.java:754)
```

### What is the expected behavior?
The `extension.yml` should have the correct opensearch host/port values and the extension initialization process should not have to update the opensearch node host and port values.

**Comment by dbwiddis** [comment]
PR #763 should fix this specific case.

**Comment by dbwiddis** [comment]
Resolved by #786.
